File: OCH_Tao/ABC/B/355.py
N,M = map(int,input().split())
A = list(map(int,input().split()))
B = list(map(int,input().split()))
C = sorted(A+B)
reach = False
for i in C:
  if i in A:
    if reach:
      print("Yes")
      break
    else:
      reach = True
  else:
    reach=False
else:
  print("No")

File: OCH_Tao/ABC/B/355.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const A = input[1].split(" ").map(x => parseInt(x, 10));
const B = input[2].split(" ").map(x => parseInt(x, 10));
const C = A.concat(B);
let flag = false;
C.sort((a, b) => a - b);
for (const i of C) {
  if (A.includes(i)) {
    if (flag) {
      console.log("Yes");
      return false;
    } else {
      flag = true;
    }
  } else {
    flag = false;
  }
}
console.log("No");